mailpoet/assets/js/src/automation/automation.tsx

```diff
@@ -1,21 +1,50 @@
 import ReactDOM from 'react-dom';
+import { useCallback, useEffect, useRef, useState } from 'react';
 import { BrowserRouter } from 'react-router-dom';
 import { TopBarWithBeamer } from 'common/top_bar/top_bar';
 import { plusIcon } from 'common/button/icon/plus';
 import { Button, Flex } from '@wordpress/components';
+import apiFetch, { APIFetchOptions } from '@wordpress/api-fetch';
 import { Workflow } from './listing/workflow';
 import { AutomationListing } from './listing';
 import { Onboarding } from './onboarding';
 import {
   CreateEmptyWorkflowButton,
   CreateWorkflowFromTemplateButton,
 } from './testing';
-import { useMutation, useQuery } from './api';
+import { ApiError, initializeApi, useMutation, useQuery } from './api';
 import { WorkflowListingNotices } from './listing/workflow-listing-notices';
+import { Notice } from '../notices/notice';
 import { MailPoet } from '../mailpoet';
 
+let errorHandler;
+
 function Content(): JSX.Element {
   const { data, loading, error } = useQuery<{ data: Workflow[] }>('workflows');
+  const noticesRef = useRef([]);
+  const [notices, setNotices] = useState(noticesRef.current);
+  const addNotice = useCallback(
+    (notice) => {
+      noticesRef.current = [notice, ...noticesRef.current];
+      setNotices(noticesRef.current);
+    },
+    [noticesRef],
+  );
+
+  const noticeErrorHandler = useCallback(
+    (apiError: ApiError) => {
+      addNotice(
+        <Notice type="error" closable timeout={false} key={Date.now()}>
+          <p>{apiError.message}</p>
+        </Notice>,
+      );
+    },
+    [addNotice],
+  );
+
+  useEffect(() => {
+    errorHandler = noticeErrorHandler;
+  }, [noticeErrorHandler]);
```

Comment:
Is there a better way of doing this without using a variable in the outer scope? It doesn't seem like I would want to register the middleware as part of the component's lifecycle because it could occur multiple times, but I could be wrong about that.

Reply:
Is it not possible to use `noticeErrorHandler` directly? The `useCallback` memoizes it. The global variable would cause problems if such a component is instantiated multiple times (which here it isn't but it's still a side-effect, I guess).

Now I realize what's the problem... you have a "local" state but want to use it in a middleware outside of the React tree... that probably won't work correctly and it may be source of some of the mentioned problems. Then the notices store would definitely help.